src/snowflake/snowpark/async_job.py

```diff
@@ -206,45 +278,84 @@ def _table_result(
             )
 
     @experimental(version="0.10.0")
-    def result(self) -> Union[List[Row], "pandas.DataFrame", Iterator[Row], int, None]:
+    def result(
+        self,
+        result_type: Optional[
+            Literal["row", "row_iterator", "pandas", "pandas_batches", "none"]
+        ] = None,
+    ) -> Union[
+        List[Row],
+        Iterator[Row],
+        "pandas.DataFrame",
+        Iterator["pandas.DataFrame"],
+        int,
+        None,
+    ]:
         """
-        Blocks and waits until the query associated with this instance finishes, then returns query results, this
-        functions acts like execute query in a synchronous way. The data type of returned results is determined by how
-        you create this :class:`AsyncJob` instance. For example, if this instance is returned by
-        :meth:`DataFrame.collect_nowait`, you will get a list of :class:`Row` s from this method.
+        Blocks and waits until the query associated with this instance finishes, then returns query
+        results, this functions acts like executing query in a synchronous way. The data type of
```

Comment:
Change 
"results, this functions acts like executing query in a synchronous way. The data type of" 
to
"results. This acts like executing query in a synchronous way. The data type of"